datasets/in/nse_debarred/crawler.py

```diff
@@ -0,0 +1,97 @@
+from typing import Generator
+from rigour.mime.types import XLS
+from normality import stringify, slugify
+from datetime import datetime
+import xlrd
+
+from zavod import Context, helpers as h
+
+SEBI_DEBARRMENT_URL = "https://nsearchives.nseindia.com/content/press/prs_ra_sebi.xls"
+OTHER_DEBARRMENT_URL = (
+    "https://nsearchives.nseindia.com/content/press/prs_ra_others.xls"
+)
+
+
+def parse_sheet(
+    sheet,
+) -> Generator[dict, None, None]:
+    headers = None
+    for row in sheet:
+        cells = []
+        for cell in row:
+            if cell.ctype == xlrd.XL_CELL_DATE:
+                # Convert Excel date format to Python datetime
+                date_value = xlrd.xldate_as_datetime(cell.value, sheet.book.datemode)
+                cells.append(date_value)
+            else:
+                cells.append(cell.value)
+        if headers is None:
+            headers = []
+            for idx, cell in enumerate(cells):
+                if not cell:
+                    cell = f"column_{idx}"
+                headers.append(slugify(cell, "_").lower())
+            continue
+
+        record = {}
+        for header, value in zip(headers, cells):
+            if isinstance(value, datetime):
+                value = value.date()
+            record[header] = stringify(value)
+        if len(record) == 0:
+            continue
+        yield record
+
+
+def crawl_item(input_dict: dict, context: Context):
+    name = input_dict.pop("entity_individual_name")
+    pan = input_dict.pop("pan")
+
+    if name is None:
+        return
+
+    entity = context.make("LegalEntity")
+    entity.id = context.make_id(name, pan)
+    entity.add("name", name)
+    entity.add("taxNumber", pan)
+    entity.add("topics", "sanction")
+    entity.add("idNumber", input_dict.pop("din_cin_of_entities_debarred"))
+
+    sanction = h.make_sanction(context, entity, key=input_dict.pop("nse_circular_no"))
```

Comment:
if it doesn't take more than 10 mins, it'd be nice to extract the circular URL and add it to sourceUrl

Reply:
From what I searched, I would need to use openpyxl to get the hyperlink. But when I try to use it I get the following error:

<img width="886" alt="Screenshot 2024-08-01 at 11 53 30" src="https://github.com/user-attachments/assets/da4224cb-7bd1-4e2b-a61c-f61e746eb6f1">